force-app/main/default/classes/RD2_ETableController.cls

```diff
@@ -129,22 +131,38 @@ public with sharing class RD2_ETableController {
     * @return List<RD2_ETableController.TableView> returns processed Recurring Donations.
     */
     private List<RD2_ETableController.TableView> getTableViews() {
-        List<RD2_ETableController.TableView> recurringDonations = new List<RD2_ETableController.TableView>();
+        Map<Id, RD2_ETableController.TableView> recurringDonations = new Map<Id,RD2_ETableController.TableView>();
         if(RD2_EnablementService.isRecurringDonations2Enabled) {
             if(controller.hasAccessToRecurringDonation()) {
+                Set<Id> fixedRecurringDonationIds = new Set<Id>();
+
                 for(npe03__Recurring_Donation__c recDonation : selector.getRecurringDonations()) {
-                    recurringDonations.add( new RD2_ETableController.TableView( recDonation,
+                    recurringDonations.put( recDonation.Id, new RD2_ETableController.TableView( recDonation,
                                                                                 controller.getFrequency(recDonation),
                                                                                 controller.getStatus(recDonation),
-                                                                                controller.getPaymentMethod(recDonation)));
+                                                                                controller.getPaymentMethod(recDonation),
+                                                                                null));
     
+                    if(recDonation.RecurringType__c == RD2_Constants.RECURRING_TYPE_FIXED) {
+                        fixedRecurringDonationIds.add(recDonation.Id);
+                    }
+                }
+
+                if(!fixedRecurringDonationIds.isEmpty()) {
```

Comment:
This method is getting a little large. Once it starts to have multiple if blocks, nested loops, etc, it's best to consider abstracting out portions of the method into new methods.

Reply:
Thanks Michael, I moved this logic to a new method.